src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -116,24 +116,29 @@ class PriorityTaggedServiceUtil
     /**
      * Gets the index defined by the default index method.
      */
-    public static function getDefaultIndex(ContainerBuilder $container, string $serviceId, string $class, string $defaultIndexMethod, string $tagName, string $indexAttribute): ?string
+    public static function getDefaultIndex(ContainerBuilder $container, string $serviceId, string $class, string $defaultIndexMethod, string $tagName, ?string $indexAttribute): ?string
     {
         if (!($r = $container->getReflectionClass($class)) || !$r->hasMethod($defaultIndexMethod)) {
             return null;
         }
 
+        $messageAttributeMissing = '';
+        if (null !== $indexAttribute) {
+            $messageAttributeMissing = sprintf(' or tag "%s" on service "%s" is missing attribute "%s"', $tagName, $serviceId, $indexAttribute);
+        }
+
         if (!($rm = $r->getMethod($defaultIndexMethod))->isStatic()) {
-            throw new InvalidArgumentException(sprintf('Either method "%s::%s()" should be static or tag "%s" on service "%s" is missing attribute "%s".', $class, $defaultIndexMethod, $tagName, $serviceId, $indexAttribute));
+            throw new InvalidArgumentException(sprintf('Either method "%s::%s()" should be static%s.', $class, $defaultIndexMethod, $messageAttributeMissing));
```

Comment:
better add the final dot in $messageAttributeMissing and concat $messageAttributeMissing to the sprintf instead of using `%s`, that will help fix fabbpot failures.

Reply:
Do you mean it like this?
```suggestion
            throw new InvalidArgumentException(sprintf('Either method "%s::%s()" should be static', $class, $defaultIndexMethod).$messageAttributeMissing);
```
And of course i would add the final dot to `$messageAttributeMissing`

Reply:
I updated it now like this :)

Reply:
Fabbot changed it then to the following:
```suggestion
            throw new InvalidArgumentException(sprintf('Either method "%s::%s()" should be static.', $class, $defaultIndexMethod).$messageAttributeMissing);
```
So the `.` would be double.